DiscordChatExporter.Domain/Discord/DiscordClient.cs

```diff
@@ -119,9 +119,10 @@ public async IAsyncEnumerable<Channel> GetGuildChannelsAsync(Snowflake guildId)
 
                 var categories = response
                     .EnumerateArray()
+                    .Where(j => j.GetProperty("type").GetInt32() == (int)ChannelType.GuildCategory)
                     .ToDictionary(
                         j => j.GetProperty("id").GetString(),
-                        j => j.GetProperty("name").GetString()
+                        j => Channel.Parse(j)
                     );
```

Comment:
Can probably do `.EnumerateArray().Select(Channel.Parse)` to avoid parsing JSON directly. Since we do it later anyway.

Reply:
If I'm understanding correctly, this fully parses every channel regardless of whether it's a category or not. Would it be worth enumerating `response` only once and grouping everything by type through the JSON? Then we could later parse categories once and then afterwards text channels once.

Reply:
Parsing itself is not an expensive operation (compared to network i/o) so if it helps us avoid duplication (i.e. referring to field names in multiple places) then it's worth it.